### PR TITLE
feat: implement PR router with LLM-based panel selection

### PR DESCRIPTION
**Pull request by moomooskycow**
## Why This Matters

Cerberus needs intelligent reviewer panel selection to minimize cost while maximizing review quality. The Python router (694 LOC) handles this today but lives outside the Elixir supervision tree. This blocks the migration path — every downstream module (ReviewSupervisor, Reviewer GenServer, Verdict) depends on the Router's output.

**Issue:** #388

## Trade-offs / Risks

- **LLM dependency for routing:** Mitigated by deterministic fallback — any LLM failure silently falls back to config-defined panel. No workflow failure possible.
- **GenServer for stateless logic:** Router holds only config reference and injected LLM function. GenServer shell is thin, logic is pure functions. Keeps supervision tree consistent.
- **Elixir-specific test patterns added:** `_test.exs` / `_test.ex` added to file classifier beyond Python parity. Low risk, correct for Elixir codebases.

## Intent Reference

> `Cerberus.Router` GenServer that parses diff, classifies model tier, calls LLM for panel selection, falls back deterministically, and outputs primary panel + reserve list.

Source: [#388 Outcome section](https://github.com/misty-step/cerberus/issues/388)

## Changes

- **`lib/cerberus/router.ex`** (525 LOC) — Full Router GenServer: diff parsing, file classification, size/tier classification, LLM routing with OpenRouter, panel validation, deterministic fallback, reserve computation
- **`test/cerberus/router_test.exs`** (398 LOC) — 35 tests covering pure functions and GenServer integration with injected LLM mocks
- **`lib/cerberus/config.ex`** — Added `model` and `enabled` fields to routing config extraction
- **`test/cerberus/config_test.exs`** — Tests for new routing config fields

## Alternatives Considered

1. **Do nothing** — Blocks entire Elixir migration. Router is the first module in the execution pipeline.
2. **Pure module (no GenServer)** — Would work functionally but breaks supervision tree pattern established by Config/Store. GenServer shell is minimal overhead.
3. **Use ReqLLM structured extraction** — Considered but plain Req + JSON schema is simpler and matches Python's approach. ReqLLM can be adopted later when Reviewer needs structured output.

## Acceptance Criteria

- [x] Router classifies PR by size bucket (small/medium/large/xlarge)
- [x] Router selects panel_size perspectives from persona bench
- [x] `trace` always included; `guard` included when code changes detected
- [x] LLM failure falls back to deterministic panel (no workflow failure)
- [x] Router output includes primary panel + reserve list
- [x] Reserve trigger conditions defined (triggering logic in Verdict module per boundaries)
- [x] `mix test test/cerberus/router_test.exs` passes

## Manual QA

```bash
cd cerberus-elixir
mix test test/cerberus/router_test.exs --trace
# Expected: 35 tests, 0 failures

mix test
# Expected: 59 tests, 0 failures
```

## What Changed

```mermaid
graph LR
    subgraph "Before"
        A[Config GenServer] --> B[Router Stub]
        B -.->|14 LOC, no logic| C[Nothing]
    end
```

```mermaid
graph LR
    subgraph "After"
        A[Config GenServer] -->|personas, routing| B[Router GenServer]
        B -->|parse diff| C[Diff Parser]
        B -->|classify| D[Size/Tier Classifier]
        B -->|LLM call| E[OpenRouter]
        B -->|fallback| F[Deterministic Panel]
        B -->|output| G["panel + reserves + tier"]
    end
```

```mermaid
sequenceDiagram
    participant Caller
    participant Router
    participant Config
    participant LLM as OpenRouter

    Caller->>Router: route(diff_text)
    Router->>Config: personas(), routing()
    Router->>Router: parse_diff, classify
    Router->>LLM: panel selection prompt
    alt LLM success
        LLM-->>Router: panel JSON
        Router->>Router: validate panel
    else LLM failure
        Router->>Router: build_fallback_panel
    end
    Router-->>Caller: {:ok, %{panel, reserves, tier, bucket}}
```

The new shape replaces a 14-LOC stub with a deep module: simple `route/3` interface, all complexity hidden behind it. Config integration is read-only. LLM transport is injectable for testing.

## Test Coverage

- `test/cerberus/router_test.exs` — 35 tests:
  - `parse_diff/1`: nil, empty, single file, multi-file, doc classification, extension histogram
  - `classify_file/1`: doc, test, code, unknown extensions
  - `classify_size/1`: all 4 boundary conditions
  - `classify_model_tier/1`: flash, standard, pro, security hints
  - `route/3` integration: panel size, valid perspectives, reserves, required perspectives, routing_used flag
  - `route/3` fallback: LLM failure, invalid response, required perspectives preserved
  - `route/3` doc-only PR: correct tier/bucket classification

## Merge Confidence

**High.** All 59 tests pass. Router is additive — no existing behavior changed. Config change is backward-compatible (new fields with defaults). LLM transport is dead code until API key is configured. DI pattern ensures full testability.

**Residual risk:** Prompt quality for panel selection untested against real LLM — will be validated when ReviewSupervisor integration lands.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Public routing API to compute reviewer panels with panel sizing, model-tier selection, and deterministic fallbacks.
  * Configuration now exposes a routing "enabled" flag and a routing "model" setting.

* **Bug Fixes**
  * Routing is more resilient: graceful handling of external routing failures with reliable fallback panels and crash safeguards.

* **Tests**
  * Extensive tests for diff parsing/classification, panel validation, config exposure, LLM integration, and fallback behavior.

* **Documentation**
  * Public API docs and observability notes added.

### CODE REVIEW

File: cerberus-elixir/lib/cerberus/router.ex [hunk 1]
```diff
@@ -1,14 +1,525 @@
...
+
+    {panel, routing_used} =
+      try_llm_routing(state.call_llm, personas, summary, panel_size, required, all_perspectives, metadata, router_model)
+
+    {panel, routing_used} =
+      if panel == [] do
+        {build_fallback_panel(routing, name_to_perspective, all_perspectives, panel_size, summary.code_changed), false}
+      else
+        {panel, routing_used}
+      end
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply cerberus-elixir/lib/cerberus/router.ex]: Fixed in 337a5a6. Applied the suggested pattern — `if Map.get(routing, :enabled, true)` gates the LLM call.

File: cerberus-elixir/lib/cerberus/router.ex [hunk 2]
```diff
@@ -1,14 +1,525 @@
...
+
+          _ ->
+            {files, current}
+        end
+
+      String.starts_with?(line, "+") and not String.starts_with?(line, "+++") and current != nil ->
+        {Map.update!(files, current, &%{&1 | additions: &1.additions + 1}), current}
+
+      String.starts_with?(line, "-") and not String.starts_with?(line, "---") and current != nil ->
+        {Map.update!(files, current, &%{&1 | deletions: &1.deletions + 1}), current}
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply cerberus-elixir/lib/cerberus/router.ex]: Fixed in 337a5a6. Both issues addressed: (1) `/dev/null` clause skips the rename, (2) guards now match `"+++ "` and `"--- "` (with trailing space) to avoid excluding content lines.

File: cerberus-elixir/lib/cerberus/router.ex [hunk 3]
```diff
@@ -1,14 +1,553 @@
...
+  Returns `{:ok, result}` where result contains:
+  - `:panel` — perspective strings for the primary panel
+  - `:reserves` — perspective strings available for escalation
+  - `:model_tier` — `:flash | :standard | :pro`
+  - `:size_bucket` — `:small | :medium | :large | :xlarge`
+  - `:routing_used` — whether LLM routing succeeded
+  """
+  def route(diff_text, opts \\ [], server \\ __MODULE__) do
+    GenServer.call(server, {:route, diff_text, opts}, 60_000)
   end
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply cerberus-elixir/lib/cerberus/router.ex]: Acknowledged — this matches the Python router parity (`router/route.py` lines 365-376 use the same hint set). Narrowing the hints is a valid follow-up but out of scope for this port. The impact is limited to selecting :pro tier (more expensive model), which is safe to over-select.
